go-controller/hybrid-overlay/pkg/controller/node_linux.go

```diff
@@ -445,59 +601,69 @@ func (n *NodeController) ensureHybridOverlayBridge() error {
 		"--", "set", "Interface", rampInt, "type=patch", "options:peer="+rampExt, "external-ids:iface-id="+portName,
 		"--", "set", "Interface", rampExt, "type=patch", "options:peer="+rampInt)
 	if err != nil {
-		return fmt.Errorf("Failed to create hybrid overlay bridge patch ports"+
+		return fmt.Errorf("failed to create hybrid overlay bridge patch ports"+
 			", stderr:%s (%v)", stderr, err)
 	}
 
-	// Add default drop rule to table 0 for easier debugging via packet counters
-	_, stderr, err = util.RunOVSOfctl("add-flow", extBridgeName, "table=0,priority=0,actions=drop")
+	// Add the VXLAN port for sending/receiving traffic from hybrid overlay nodes
+	_, stderr, err = util.RunOVSVsctl("--may-exist", "add-port", extBridgeName, extVXLANName,
+		"--", "set", "interface", extVXLANName, "type=vxlan", `options:remote_ip="flow"`, `options:key="flow"`)
 	if err != nil {
-		return fmt.Errorf("failed to set up hybrid overlay bridge default drop rule,"+
-			"stderr: %q, error: %v", stderr, err)
+		return fmt.Errorf("failed to add VXLAN port for ovs bridge %s"+
+			", stderr:%s: %v", extBridgeName, stderr, err)
 	}
 
-	// Handle ARP for gateway address internally
-	portMACRaw := strings.Replace(portMAC.String(), ":", "", -1)
-	portIPRaw := getIPAsHexString(portIP)
-	_, stderr, err = util.RunOVSOfctl("add-flow", extBridgeName,
-		fmt.Sprintf("table=0,priority=100,in_port=%s,arp,arp_tpa=%s,"+
+	flows := make([]string, 0, 10)
+	// Add default drop rule to tables for easier debugging via packet counters
+	for _, table := range []int{0, 1, 10, 20} {
+		flows = append(flows, fmt.Sprintf("table=%d,priority=0,actions=drop", table))
+	}
+	// Handle ARP for gateway address internally towards pods
+	// resubmit to table 1 for gateway mode arp processing
+	portMACRaw := strings.Replace(n.drMAC.String(), ":", "", -1)
+	portIPRaw := getIPAsHexString(n.drIP)
```

Comment:
You shouldn't need the "raw" forms if you use `set_field:%s->arp_sha, set_field:%s->arp_spa` rather than `load:0x%s->NXM_NX_ARP_SHA[], load:0x%s->NXM_OF_ARP_SPA[]`

Reply:
you are correct in most cases. I tried to actually change where i saw load to set_field where I noticed this (don't think I caught them all). However, we need to use load in learn actions. Learn actions only support load.